src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -864,16 +875,22 @@ impl Coordinator {
             let ts = determination.timestamp_context.timestamp_or_default();
             let mut transitive_storage_deps = BTreeSet::new();
             let mut transitive_compute_deps = BTreeSet::new();
-            for id in id_bundle
+            for item_id in id_bundle
                 .iter()
+                .map(|gid| self.catalog.state().get_entry_by_global_id(&gid).id())
                 .flat_map(|id| self.catalog.state().transitive_uses(id))
             {
-                match self.catalog.state().get_entry(&id).item() {
+                let entry = self.catalog.state().get_entry(&item_id);
+                match entry.item() {
+                    // TODO(parkmycar): Adding all of the GlobalIds an object depends on is
+                    // probably incorrect, but it's okay for now since the only thing that can have
+                    // multiple GlobalIds are Tables. In the future we should track dependencies
+                    // based on `GlobalId` or (`CatalogItemId`, `Version`).
```

Comment:
I'm having trouble parsing this comment. Using GlobalIds currently is incorrect, but in the future we should track GlobalIds. Is there a typo somewhere?

Reply:
Re-worded this, basically we don't want to depend on _all_ GlobalIds associated with an item, e.g. all versions of a table, just the versions the peek actually depends on